MobiusExtras/Source/MobiusViewController.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import MobiusCore
+import UIKit
+
+public extension UIViewController {
+    /// Attach a `MobiusController` to this `UIViewController`.
+    /// A `Connection` is returned which can be used to post events to the loop.
+    /// The `MobiusController` will start when this function is called, and stop when this `UIViewController` is
+    /// initialized or when the returned `Connection` is disposed.
+    ///
+    /// Note: You should not be calling `start`, `connectView`, `stop`, or `disconnectView` on the `MobiusController`
+    /// when using this extension.
+    ///
+    /// Note: The `MobiusController` will be placed in a `UIViewController` and added as a child `UIViewController` to
+    ///  this `UIViewController`.
+    ///
+    /// - Parameter controller: The `MobiusController` that shuold be attached
+    /// - Parameter onModelChange: A closure which is called whenever the loop's model changes.
+    func useMobius<Model, Event, Effect>(
+        controller: MobiusController<Model, Event, Effect>,
+        modelChanged onModelChange: @escaping (Model) -> Void
+    ) -> Connection<Event> {
+        let holder = MobiusHolder(controller: controller, onModelChange: onModelChange)
+        addChild(holder)
+        holder.didMove(toParent: self)
+        return Connection(
+            acceptClosure: { [unowned holder] event in
+                holder.handleEvent(event)
+            },
+            disposeClosure: { [unowned holder] in
+                holder.willMove(toParent: nil)
+                holder.removeFromParent()
+            }
+        )
+    }
+}
+
+private class MobiusHolder<Model, Event, Effect>: UIViewController {
```

Comment:
Feels weird, but I suppose I have heard of other cases of using a UIViewController which does not actually control a view. I assume you've already ruled out using a PONSO and `objc_setAssociatedObject()`?

Reply:
That's a good idea, thanks. My initial idea was to use the lifecycle hooks in the child ViewController, but that turned out not to be necessary.